go/api/base_client.go

```diff
@@ -1905,3 +1905,96 @@ func (client *baseClient) ZScanWithOptions(
 	}
 	return handleScanResponse(result)
 }
+
+// Returns stream message summary information for pending messages matching a stream and group.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	key - The key of the stream.
+//	group - The consumer group name.
+//
+// Return value:
+// An XPendingSummary struct that includes a summary with the following fields:
+//
+//	NumOfMessages: The total number of pending messages for this consumer group.
+//	StartId: The smallest ID among the pending messages or nil if no pending messages exist.
+//	EndId: The greatest ID among the pending messages or nil if no pending messages exists.
+//	GroupConsumers: An array of ConsumerPendingMessages with the following fields:
+//	  ConsumerName: The name of the consumer.
+//	  MessageCount: The number of pending messages for this consumer.
+//
+// Example
+//
+//	result, err := client.XPending("myStream", "myGroup")
+//	if err != nil {
+//	  return err
+//	}
+//	fmt.Println("Number of pending messages: ", result.NumOfMessages)
+//	fmt.Println("Start and End ID of messages: ", result.StartId, result.EndId)
+//	for _, consumer := range result.ConsumerMessages {
+//	  fmt.Printf("Consumer messages:  %s: $v\n", consumer.ConsumerName, consumer.MessageCount)
+//	}
+//
+// [valkey.io]: https://valkey.io/commands/xpending/
+func (client *baseClient) XPending(key string, group string) (XPendingSummary, error) {
```

Comment:
TBD should we return `XPendingSummary` or `*XPendingSummary`?

Reply:
Based on the recommendations in https://go.dev/wiki/CodeReviewComments and https://stackoverflow.com/questions/23542989/pointers-vs-values-in-parameters-and-return-values I believe the appropriate response should be to return the struct rather than a pointer.  This is a small struct.  The only potential large content would be the slice of Consumer Pending Messages which is already a pointer under the covers.